src/utils.c

```diff
@@ -221,43 +221,23 @@ ts_time_value_to_internal(Datum time_val, Oid type_oid, bool failure_ok)
 int64
 ts_interval_from_now_to_internal(Datum interval, Oid type_oid)
 {
-	TimestampTz now;
-	Datum		res;
+	Datum		res = TimestampTzGetDatum(GetCurrentTimestamp());
 
 	/*
-	 * This is really confusing but looks like it is how postgres works. Event
-	 * though there is a function called Datum now() that calls
-	 * GetCurrentTransactionStartTimestamp and returns TimestampTz, that is
-	 * not the same as now() function in SQL and even though return type is
-	 * Timestamp WITH TIMEZONE, GetCurrentTransactionStartTimestamp does not
-	 * incorporate current session timezone infromation in the returned value.
-	 * In order to take this timezone value set by the user into account, I
-	 * convert timestamp to POSIX time structure using timestamp2tm *which has
-	 * access to current timezone* setting through session_timezone variable
-	 * and incorporates it in the returned structure. I then convert it back
-	 * to TimestampTz through a similar function which takes this timezone
-	 * information int account.
+	 * GetCurrentTimestamp returns timestamp in UTC so we need to adjust to
+	 * session timezone for all datatypes except timestamptz
 	 */
-	int			tzoff;
-	struct pg_tm tm;
-	fsec_t		fsec;
-	const char *tzn;
-
-	timestamp2tm(GetCurrentTransactionStartTimestamp(),
-				 &tzoff, &tm, &fsec, &tzn, NULL);
-	tm2timestamp(&tm, fsec, NULL, &now);
+	if (TIMESTAMPTZOID != type_oid)
```

Comment:
I think this could use a comment like
```suggestion
	/*
	 * Postgres uses the same functions for TIMESTAMPTZ and regular TIMESTAMPs,
	 * (see ${link to where this is documented} for more detail)
	 * so if we're returning a time in the local timezone we simply need to convert to that 
	 * timezone here.
	 */
	if (TIMESTAMPTZOID != type_oid)
```